File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {
    public int countSubstrings(String s) {
        int count=0;
        for(int i=0;i<s.length();i++){
            for(int j=i;j<s.length();j++){
                if(isPalindromic(s.substring(i,j+1))){
                    count++;
                }
            }
        }
        return count;
    }
    public static boolean isPalindromic(String str){
        int left=0;
        int right=str.length()-1;
        while(left<=right){
            if(str.charAt(left)!=str.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: 0647-palindromic-substrings/0647-palindromic-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countSubstrings = function(s) {
    let count=0;
    for(let i=0;i<s.length;i++){
        for(let j=i;j<s.length;j++){
            if(isPalindromic(s.substring(i,j+1))){
                count++;
            }
        }
    }
    return count;
};
var isPalindromic=function(s){
    let left=0;
    let right=s.length-1
    while(left<=right){
        if(s.charAt(left)!=s.charAt(right)){
            return false;
        }
        left++;
        right--;
    }
    return true;
};